### PR TITLE
azure: always infer tenant_id

### LINKED ISSUE #249: azure: always infer the tenant_id

### PR DESCRIPTION
**Pull request by colemickens**
Fixes #249.

(This stacks on top of PR #248 and should be rebased and merged only after #248 is merged).

cc: @ahmetalpbalkan

### CODE REVIEW

File: phase1/azure/do [hunk 1]
```diff
@@ -67,13 +61,46 @@ check_auth() {
   fi
 }
 
+check_tenant_id() {
+  subscription_id="$(jq -r '.phase1.azure.subscription_id' ${configjson})"
+  tenant_id="$(curl https://management.azure.com/subscriptions/${subscription_id}?api-version=2016-01-01 -s -D - -o /dev/null | grep WWW-Authenticate | egrep '[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}' -o)"
```
Comment (ahmetalpbalkan) [inline phase1/azure/do]: add `-f` to curl

Comment (colemickens) [inline phase1/azure/do]: done

Comment (colemickens) [inline phase1/azure/do]: Actually... that's not good. The http response is an error, so this breaks things.

Comment (ahmetalpbalkan) [inline phase1/azure/do]: ah, right...

**Pull request merged**